openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -662,27 +659,24 @@ def create_entity_list_with_rows(
     # Do a quick check that the given entities are in the right learning package:
     if learning_package_id:
         if PublishableEntity.objects.filter(
-            pk__in=entity_pks,
+            pk__in=[entity.entity_pk for entity in entity_rows],
         ).exclude(
             learning_package_id=learning_package_id,
         ).exists():
             raise ValidationError("Container entities must be from the same learning package.")
 
```

Comment:
In https://github.com/openedx/openedx-learning/pull/278#discussion_r1994432451 @bradenmacdonald said:

>  I'd like to also introduce some sort of verification that the entity version objects actually match the corresponding entity objects

But I was unable to come up with a simple query to do this with a list of entity + version PKs. Ideas welcome!

Reply:
I think it needs to be a separate query.

Can we do something like `PublishableEntityVersion.objects.filter(pk__in=versions_list).values_list('entity_id', flat=True)` to retrieve the IDs, then compare that to the list of entity IDs we received? (After filtering out 'unpinned' ones where no version ID is given)

Reply:
Oh of course! Thanks @bradenmacdonald , see 8422317354ffc530c8b6374a72a0ff6595469700.